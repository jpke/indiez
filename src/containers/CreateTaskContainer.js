import React from 'react';
import {connect} from 'react-redux';
import {updateNewTask, submitTask, createTask} from '../actions/taskActions';
import CreateTaskView from '../components/CreateTaskView';

//Container for CreateTaskView
export const CreateTaskContainer = (props) => {

  const createTask = () => {
    props.submitTask(props.newTask)
  }

  return (
      <CreateTaskView
        updateNewTask={props.updateNewTask}
        createTask={createTask}
        newTask={props.newTask}
        cancelCreateTask={props.cancelCreateTask}

      />
  );
};

// QuizPage.propTypes

function mapStateToProps(state) {
  return {
      newTask: state.newTask
    }
}

function mapDispatchToProps(dispatch) {
  return {
    updateNewTask: (key, value) => dispatch(updateNewTask(key, value)),
    submitTask: (newTask) => dispatch(submitTask(newTask)),
    cancelCreateTask: () => dispatch(createTask())
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CreateTaskContainer);
