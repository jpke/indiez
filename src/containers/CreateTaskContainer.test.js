/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
const store = configureStore();

import CreateTaskContainer from './CreateTaskContainer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <CreateTaskContainer />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
