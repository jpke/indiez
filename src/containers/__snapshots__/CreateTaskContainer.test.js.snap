exports[`test renders correctly 1`] = `
<div
  id="newTaskForm">
  <h2>
    Create New Task
  </h2>
  <form
    onSubmit={[Function]}>
    <label>
      Task Name
    </label>
    <input
      defaultValue="task name"
      name="name"
      onChange={[Function]}
      type="text" />
    <label>
      End Date
    </label>
    <input
      defaultValue="2017-04-19"
      name="endDate"
      onChange={[Function]}
      type="date" />
    <label>
      Description
    </label>
    <textarea
      defaultValue="task description"
      name="description"
      onChange={[Function]}
      type="text" />
    <section
      className="taskButtons">
      <button
        onClick={[Function]}>
        Cancel
      </button>
      <button
        type="submit">
        Create
      </button>
    </section>
  </form>
</div>
`;
